### PR TITLE
[PCM-368] Add check before doing addTrack.

### PR DESCRIPTION
**Pull request by jbelmont**
I was reading through [Adapter Js Repo](https://github.com/webrtc/adapter/search?utf8=%E2%9C%93&q=The+adapter.js+polyfill&type=) and saw that there is a check for *addTrack* to be added once.

Here is one file in [Adapter.js](https://github.com/webrtc/adapter/blob/b94fc1fd92fb184a45c5af778b2483634803e51d/release/adapter.js) line 2673. I added the check in 3 different spots because addTrack is used in AudioBandwidth, VideoBandwidth, and in CameraResolutionTest

### CODE REVIEW

**Comment by xdumaine** [comment]
@jbelmont I pulled this down and ran it and I still got the error.

**Comment by jbelmont** [comment]
@xdumaine I ran it locally and made an explicit conditional checks before calling addTrack but I don't see the error. 

My Following Steps:

1. `yarn link` in webrtc-troubleshooter
2. `yarn link "webrtc-troubleshoot"` in ember-webrtc-troubleshoot
3. Ran `yarn install` in ember-webrtc-troubleshoot for newly linked module
4. Ran `yarn start` and opened up [http://localhost:4200/](http://localhost:4200/)
5. I saw some errors but not the adapter.js addTrack polyfill Error

**Comment by jbelmont** [comment]
Some of the errors about `addTrack` not being defined are gone now

**Comment by xdumaine** [comment]
@jbelmont you don't need to use the ember repo to reproduce an error in this one. Just serve up the test page in this repo.

**Comment by jbelmont** [comment]
@xdumaine I served the test page and I got the following in console:

**I Skipped some console output**

- *Video stream passed*
- *SUCCESS: Captured video using expected resolution.*
- *SUCCESS: Captured video using expected resolution.*
- *SUCCESS: Captured video using expected resolution.*
- *SUCCESS: Supported: 160x120*
- Starting Bandwidth Test
- test.js:69 test failure TypeError: Failed to construct 'RTCPeerConnection': cannot convert to dictionary.

**Comment by xdumaine** [comment]
Ok... have you built with latest deps? What browser are you running? Version? I'm in latest stable chrome and I get the error described in the JIRA.

**Comment by jbelmont** [comment]
Okay once I ran `yarn upgrade` I was able to reproduce the error. I have been running in Chrome Beta (Version 62), but I just tried Chrome Stable (Which is on Version 62 as well) and was able to reproduce the error, so obviously I haven't fixed it yet, will keep working on it.

File: src/utils/TestSuite.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export default class TestSuite {
   start () {
     return new Promise((resolve, reject) => {
       return this.runNextTest().then(() => {
+        this.results = [];
```
Comment (jbelmont) [inline src/utils/TestSuite.js]: I think cleaning up this reference was the issue

**Comment by jbelmont** [comment]
@xdumaine I tested it in Chrome Beta and Chrome Stable just now and both are in version 62 and the only thing that I see now is test.js:69 test failure TypeError: Failed to construct 'RTCPeerConnection': cannot convert to dictionary but everything else seems okay

File: src/utils/TestSuite.js (hunk 1 above)
Comment (xdumaine) [reply src/utils/TestSuite.js]: I'm not sure I understand. Why would you empty out the results?

Comment (jbelmont) [reply src/utils/TestSuite.js]: Well doesn't `this.results` instance get shared between the AdvancedCameraTest, AudioBandwidthTest, and the CameraResolutionTest, shouldn't the instance be cleaned up before each type of test. Once I cleared it, the test failures went away

**Comment by xdumaine** [comment]
Build failed because you're referencing our internal artifactory npm. Need to resolve yarn.lock without that.

File: src/utils/TestSuite.js (hunk 1 above)
Comment (xdumaine) [reply src/utils/TestSuite.js]: The results are populated by the test, and you're emptying the results as soon as the test finishes.

**Comment by xdumaine** [comment]
I merged this by accident. Just pushed revert.

**Pull request merged**